src/js/letters/components/Address.jsx

```diff
@@ -76,32 +77,42 @@ class Address extends React.Component {
           label="Street address"
           name="address"
           autocomplete="street-address"
-          charMax={30}
+          charMax={35}
           value={this.props.address.addressOne}
           required={this.props.required}
           onValueChange={(update) => this.props.onInput('addressOne', update)}/>
         <ErrorableTextInput
           label="Street address (optional)"
           name="address"
           autocomplete="street-address"
-          charMax={30}
+          charMax={35}
           value={this.props.address.addressTwo}
           onValueChange={(update) => this.props.onInput('addressTwo', update)}/>
         <ErrorableTextInput
           label="Street address (optional)"
           name="address"
           autocomplete="street-address"
-          charMax={30}
+          charMax={35}
           value={this.props.address.addressThree}
           onValueChange={(update) => this.props.onInput('addressThree', update)}/>
-        <ErrorableTextInput errorMessage={errorMessages.city}
-          label={<span>City <em>(or APO/FPO/DPO)</em></span>}
-          name="city"
-          autocomplete="address-level2"
-          charMax={30}
-          value={this.props.address.city}
-          required={this.props.required}
-          onValueChange={(update) => this.props.onInput('city', update)}/>
+        {/* Text field for non-military, select box for military */}
```

Comment:
Hm, will this work? I'm afraid it's going to put users on a weird loop they can't get out of. Let's say they have a defaulted military address, but they want to change it to a regular domestic address. The only options they'll see for the city are the military ones, because their address is defaulted to military, which in turn is based on if a military city or state is selected. And the only options they'll see for the state are the military ones too because those are filtered based on the presence of a military city. So how will they be able to change those fields?

Reply:
Hmm, you make an excellent point... So it _is_ possible to change back to a domestic address, but it's not intuitive.

The problem as it stands right now:
1. To make an address military, the state selection needs to be AA / AE/ AP
2 When the address is military, the city becomes a dropdown where only military post office type codes are displayed
3. When a military "city" is chosen, the state list is restricted to AA / AE / AP
4. This creates a situation where I can't select a "regular" state to change the address type to domestic

The workaround is to select a blank city before changing the state, but that's not really acceptable, is it?

Even without changing it to a dropdown, we'd still have to clear out the city field to expand the list of states to include "regular" states, so this isn't new with this PR, but...still a problem.

I think we've got a couple options:
1. Not limit the states in the dropdown to AA / AE / AP if the city is military
    - The trouble with this is they're likely to enter the city first, which may be...misleading
2. Not have a dropdown for the city
    - They'd have to type out the military "city" name instead of selecting it from a list, but by the time they get to state, they could select whatever state they're in

For now, I'll change it back to a text field and write a ticket with ☝️ in it for @goldenmeanie to decide what's best.

Reply:
I agree we need input from @goldenmeanie as this is a tough one and I'm not convinced we've nailed a good user experience yet